### ISSUE TITLE
Page tree suggestion

### ISSUE THREAD

**Issue by hdrei** [Lumi-supercomputer/lumi-self-learning#24]
# Page tree

Suggestion for the page tree (to start working with). This can of course be modified as we go, as our knowledge increases of what is good/needed. Please leave comments especially about the main chapters (if there are unnecessary ones, if something is missing, or if some of them should maybe be combined). 

*If others agree with the page tree and the following points, this should be moved out of the issues (somewhere else, where?).*

- You can give just an idea of a tutorial (by writing the tutorial name in the page tree) without starting to work with it yourself. (Should these ideas be confirmed somehow, or should we just go freely with creating all kinds of tutorials one would think of are useful?) I added some tutorial-ideas, but haven't fully been thinking through what is needed/necessary.

- Tables in markdown don't support expanding a cell on multiple lines, so let's not try to do a table of this, in the end. One could write their name after a tutorial they are working with currently (to avoid many people doing the same work), and the name could be maybe marked in bold (using of colors isn't probably possible). 

- What should be the protocol of adding content to the package structure? A PR - who to mark as a reviewer? Maybe at least the other people who are currently working with the same chapter?

- Some of the tutorials to be marked as "essential", and some to be marked as "extra", "advanced", or something else?

- In all cases, *do not rewrite content that already exists* especially in "LUMI documentation" or in "LUMI training material archive". If the already existing material suits our purposes with this material package, let's link to the existing material.

------------------------------------------------------------------------------------------------------------------

## LUMI Self Learning material

**1. Motivation**: Why to use supercomputing and LUMI (Benefits of HPC, what is a supercomputer and what is it not, comparing supercomputing cluster to cloud computing platforms (?), examples of what can be achieved on LUMI what it comes to performance etc)
	
**2. Prerequisites** (accounts, connecting)
	Info(?): How to apply (and who can apply resources) and get a user account (links to www.lumi-supercomputer.eu and lumi docs of what is relevant - not re-writing the existing info)
	Info(?): Accepting an invitation to a LUMI project (links to relevant video material and Puhuri docs)
	Tutorial: Create ssh-keys in terminal (Linux, Mac, Windows OpenSSH)
        Tutorial: Create ssh-keys with PuTTY/MobaXterm
	Tutorial: Register your public key in MyAccessID
	Tutorial: Register your public key in MyCSC
	Tutorial: Connect to LUMI via terminal 
	Tutorial: Connect to LUMI with PuTTY/?
	Tutorial: Connect to LUMI with OpenSSH?
	Tutorial: Connect to LUMI via LUMI web interface
	Troubleshooting? (To be included in the individual tutorials?)

**3. Introduction to LUMI environment** (~ cluster.md in HPC carpentry material, checking your project info with "lumi-workspaces" LUMI HW (very briefly, mainly "why should you know about this", and links to LUMI docs? and LUMI training material archive, where there is a good description about this.)
	Tutorial: After connecting -  Login nodes, etc
	Tutorial: 
	
**4. Disk areas** (Navigating and using the main disk areas, as in "CSC comp env" material.)
	Tutorial: Main disk areas on LUMI
	Tutorial: Finding out where you have a lot of data (?) ("du -sh *" is good to use on lumi, unlike the Finnish Puhti)
	Tutorial: Fast disk areas on LUMI
	
**5. Modules and software stacks**
	Tutorial: LMOD1: Loading modules (module list, module avail, module spider, module load, module unload, ...)
	Tutorial: Software stacks (LUMI, CrayEnv, partitions)
	Tutorial: LMOD2 (?): Loading and unloading environments (module purge, module --force unload LUMI, ..)
	Extra tutorial: Saving your module environment (?)
	Tutorial: Finding available & installable software
		(LUMI docs https://lumi-supercomputer.github.io/LUMI-EasyBuild-docs/ , local sw collections, checking pre-installed and user-installable software)
 	Tutorial: Using pre-installed software (?)

**7. Installing software**	
	Tutorial: Installing user-installable software with EasyBuild
	Tutorial: Python packages on LUMI (maybe several tutorials related to this?)
	Tutorial: Pytorch (from *Georg Z*?)
	Tutorial: Tensorflow (from *Georg Z*?)
	Tutorial: (Some R tutorial?)
	Tutorial: Using LUMI container wrapper to install software
	Tutorial: Installing software with Spack 
	Tutorial: Creating an own own module for your installed software (?)
	Tutorial: Adding own EasyBuild recipes

**8. Cray environment on LUMI** (?) (Should we have a section about this? Or, just a mention somewhere, and link to LUMI docs and LUMI training material archive?)
	Tutorial/info: Compilers on LUMI (https://docs.lumi-supercomputer.eu/development/compiling/prgenv/)
	Tutorial: Using Cray libraries (?)
	Tutorial: 

**9. Scheduler and batch jobs**

**10. Using resources efficiently**
	Tutorial: Estimating required resources (?)
	Tutorial: (~ responsibility.md from HPC carpentry? login nodes as shared resource etc)

**11. Managing data and transferring files** (An overlap with "Disk areas" chapter?)
	Tutorial: 
	Tutorial: LUMI-O object storage
	Tutorial: Moving large amounts of data (?)
	Tutorial: 

**12. Working with containers**
	Tutorial: 
	Tutorial: Using Cotainr on LUMI
	Tutorial: 
	
**13. Advanced topics** (e.g. high-throughput workflows, CPU-GPU binding)
